### PR TITLE
Staging sartura

### PR DESCRIPTION
**Pull request by dfranusic**
- there are more fixes coming, but this is a start and let's take it from here
- we were unable to test it on any other device other than ecw5211
- a couple of ecw5410's are currently being prepared for firmware flashing and testing

### CODE REVIEW

File: feeds/wlan-ap/opensync/patches/11-protobuf-definitions [hunk 1]
```diff
@@ -26,7 +26,7 @@ Index: opensync-2.0.5.0/interfaces/opensync_stats.proto
 +    CLIENT_DISCONNECT                       = 2;
 +    CLIENT_FAILURE                          = 3;
 +    CLIENT_FIRST_DATA                       = 4;
-+    CLIEND_ID                               = 5;
++    CLIENT_ID                               = 5;
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/11-protobuf-definitions:29]: unrelated change

File: feeds/wifi-trunk/hostapd/files/hostapd.sh [hunk 2]
```diff
@@ -261,8 +261,6 @@ hostapd_common_add_bss_config() {
 	
 	config_add_string 'owe_transition_bssid:macaddr' 'owe_transition_ssid:string'
 
-	config_add_int beacon_rate
-	config_add_int rssi_reject_assoc_rssi
```
Comment (blogic) [inline feeds/wifi-trunk/hostapd/files/hostapd.sh:265]: this looks odd, I would assume that just not adding the uci would be enough.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 3]
```diff
@@ -0,0 +1,1489 @@
+Index: opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_events.h
+===================================================================
+--- /dev/null
++++ opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_events.h
+@@ -0,0 +1,364 @@
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:5]: new files should go into the src folder and not the patches folder

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 4]
```diff
@@ -0,0 +1,1489 @@
...
++{
++    return dppline_put(DPP_T_EVENTS, rpt);
++}
++
++/*
+  * Create the protobuf buff and copy it to given buffer
+  */
+ #ifndef DPP_FAST_PACK
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/inc/target_OPENWRT.h
+===================================================================
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:698]: dont patches files in the src/ folder but apply the patch to the actual file please

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 5]
```diff
@@ -0,0 +1,1489 @@
...
++	if (nl80211_rt_events_handler(&nl_call_param) < 0)
++		ret = false;
++
++        return ret;
++}
++
+ 
+ /******************************************************************************
+  *  CLIENT definitions
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/src/stats_nl80211.c
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:775]: dont patches files in the src/ folder but apply the patch to the actual file please

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 6]
```diff
@@ -0,0 +1,1489 @@
...
+ typedef void target_capacity_data_t;
+ 
++typedef void target_event_record_t;
++
+ #include "target_common.h"
+ 
+ #endif  /* TARGET_NATIVE_H_INCLUDED */
+Index: opensync-2.0.5.0/vendor/tip/src/lib/target/inc/target_tip.h
+===================================================================
+--- opensync-2.0.5.0.orig/vendor/tip/src/lib/target/inc/target_tip.h
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:889]: dont patches files in the src/ folder but apply the patch to the actual file please

**Comment by dfranusic** [comment]
Hi Mike,

No problem. I've sent you the updated version on Slack. I expect more fixes coming (unrelated to proto files), but I will make sure to keep the cloud guys informed should I need to make more changes to the opensync_stats.proto

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 7]
```diff
@@ -0,0 +1,1489 @@
...
+     STS_REPORT_MAX,
+     STS_REPORT_ERROR = STS_REPORT_MAX
+ } sm_report_type_t;
+Index: opensync-2.0.5.0/src/sm/src/sm_events_report.c
+===================================================================
+--- /dev/null
++++ opensync-2.0.5.0/src/sm/src/sm_events_report.c
+@@ -0,0 +1,238 @@
++/*
++Copyright (c) 2015, Plume Design Inc. All rights reserved.
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:962]: is this file from Plume ? if not add bsd 3clause SPDX instead

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 8]
```diff
@@ -0,0 +1,1489 @@
...
+ 				int dwell_time, radio_scan_type_t scan_type,
+ 				target_scan_cb_t *scan_cb, void *scan_ctx);
+ extern int nl80211_scan_abort(struct nl_call_param *nl_call_param);
+ extern int nl80211_scan_dump(struct nl_call_param *nl_call_param);
++// real time events handler used by sm_events_report
++extern int nl80211_rt_events_handler(struct nl_call_param *nl_call_param);
++
+ 
+ #endif
+Index: opensync-2.0.5.0/platform/openwrt/src/lib/target/src/radio_nl80211.c
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1260]: and here aswell

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 9]
```diff
@@ -0,0 +1,1489 @@
...
++    memcpy(cap->sta_mac, nla_data(mac), ETH_ALEN);
++    memcpy(cap->ssid, nla_data(ssid), nla_len(ssid));
++    cap->session_id = get_timestamp();
++    // add to global list
++    ds_dlist_insert_tail(&g_rt_events.event_list, event_entry);
++}
++
++static void mlme_ev_disconnect(struct nl_msg *msg, struct nlattr **tb)
++{
++    // check ssid and mac
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1337]: // c++ commnt,
/* c99 comment */
also please follow the code style of the rest of the code in the file. it uses tabs for indenting

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 10]
```diff
@@ -0,0 +1,1489 @@
...
++    cap->session_id = get_timestamp();
++    // add to global list
++    ds_dlist_insert_tail(&g_rt_events.event_list, event_entry);
++}
++
++
++
++static void mlme_ev_recv(struct nl_msg *msg, struct nlattr **tb)
++{
++    //const uint8_t *data;
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1360]: dead code can be removed

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 11]
```diff
@@ -0,0 +1,1489 @@
...
++    size_t len;
++    struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
++    struct nlattr *frame = tb[NL80211_ATTR_FRAME];
++    struct nlattr *addr = tb[NL80211_ATTR_MAC];
++
++    // frame should be present
++    if(!(frame && addr)) return;
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1367]: line break + indent between clause and return

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 12]
```diff
@@ -0,0 +1,1489 @@
...
++    case NL80211_CMD_ASSOCIATE:
++            mlme_ev_assoc(msg, tb);
++            break;
++    case NL80211_CMD_DEAUTHENTICATE:
++    case NL80211_CMD_DISASSOCIATE:
++    case NL80211_CMD_UNPROT_DEAUTHENTICATE:
++    case NL80211_CMD_UNPROT_DISASSOCIATE:
++            mlme_ev_disconnect(msg, tb);
++            break;
++
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1388]: stray blank line

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 13]
```diff
@@ -0,0 +1,1489 @@
...
++    }
++
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1390]: stray blank line

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 14]
```diff
@@ -0,0 +1,1489 @@
...
+ static int nl80211_recv(struct nl_msg *msg, void *arg)
+ {
+ 	struct genlmsghdr *gnlh = nlmsg_data(nlmsg_hdr(msg));
+@@ -433,6 +537,23 @@ static int nl80211_recv(struct nl_msg *m
+ 	case NL80211_CMD_GET_WIPHY:
+ 		nl80211_add_phy(tb, phyname);
+ 		break;
++
++        // connect
++        case NL80211_CMD_ASSOCIATE:
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1402]: tab vs space

**Review by blogic — CHANGES_REQUESTED** [review]

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 15]
```diff
@@ -433,8 +433,8 @@ Index: opensync-2.0.5.0/src/lib/datapipeline/src/dppline.c
          }
  
          free(s);
-@@ -623,6 +641,40 @@ static bool dppline_copysts(dppline_stat
-                 dst->u.network_probe.timestamp_ms = report_data->timestamp_ms;
+@@ -624,6 +642,40 @@ static bool dppline_copysts(dppline_stat
+ 		dst->u.network_probe.timestamp_ms = report_data->timestamp_ms;
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:437]: untrelated whitespace change

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 16]
```diff
@@ -578,10 +578,14 @@ Index: opensync-2.0.5.0/src/lib/datapipeline/src/dppline.c
...
-@@ -1238,6 +1387,81 @@ static void dppline_add_stat_client(Sts_
-             sizeof(*s->u.device), s->size, size);*/
- }
- 
+@@ -1235,8 +1384,87 @@ static void dppline_add_stat_client(Sts_
+             size += n * sizeof(*dtid->sojourn);
+         }
+     }
+-    /*LOG(DEBUG, "============= %s size raw: %d alloc: %d proto struct: %d", __FUNCTION__,
+-            sizeof(*s->u.device), s->size, size);*/
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:586]: random removal of code

**Review by blogic — CHANGES_REQUESTED** [review]
this patch seems to patch a previous patch. the PR is quite heavy on bad code formatting. please consider fixing the comments I made not just where I made them but throughout the entire patch and repost.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 7 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:962]: No, this is one of mine. It was a copy/paste from the one from Plume, hence the wrong licence.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch [hunk 17]
```diff
@@ -0,0 +1,1489 @@
...
++#include "ds.h"
++#include "ds_dlist.h"
++
++#include "dpp_types.h"
++
++// string limits
++#define DPP_REASON_STR_LEN  129
++#define DPP_CLT_ID_LEN      129
++
++// proto: EventType
```
Comment (blogic) [inline feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:18]: please consider using /* c99 */ and not // cpp

**Comment by dfranusic** [comment]
The code formatting issues are related to my **.clang-format** file which I use in **Vim**. Is there an official one for this project? If there isn't, I'll fix it manually.

**Comment by blogic** [comment]
We try to follow the kernel coding style for the osync2.0 code that we write. Thanks !

File: feeds/wifi-trunk/hostapd/files/hostapd.sh (hunk 2 above)
Comment (dfranusic) [reply feeds/wifi-trunk/hostapd/files/hostapd.sh:265]: This was causing problems on ecw5211, as mentioned by @RickSommerville. The problematic patch was introduced in one of the previous commits, and it was promptly followed by another one which was supposed to undo the changes. Apparently, as mentioned by @RickSommerville, the fix was incomplete. I removed those lines in order to prevent **hostapd** from crashing on **ecw5211**.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 14 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1402]: The eternal battle of tabs vs spaces. Again, my Vim setup.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 16 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:586]: Sorry about that.

**Comment by dfranusic** [comment]
Thanks for your extensive feedback @blogic. I'll go through this and push another commit which will, hopefully, fix all these issues.

File: feeds/wifi-trunk/hostapd/files/hostapd.sh (hunk 2 above)
Comment (ArifAlam) [reply feeds/wifi-trunk/hostapd/files/hostapd.sh:265]: Hi, I don't agree with removing rssi_reject_assoc_rssi.  Why is rssi_reject_assoc_rssi getting removed?

File: feeds/wifi-trunk/hostapd/files/hostapd.sh [hunk 18]
```diff
@@ -361,7 +358,6 @@ hostapd_set_bss_options() {
 	append bss_conf "uapsd_advertisement_enabled=$uapsd" "$N"
 	append bss_conf "utf8_ssid=$utf8_ssid" "$N"
 	append bss_conf "multi_ap=$multi_ap" "$N"
-	append bss_conf "rssi_reject_assoc_rssi=$rssi_reject_assoc_rssi" "$N"
```
Comment (ArifAlam) [inline feeds/wifi-trunk/hostapd/files/hostapd.sh:364]: Same as above... why is rssi_reject_assoc_rssi getting removed?

File: feeds/wifi-trunk/hostapd/files/hostapd.sh [hunk 19]
```diff
@@ -338,7 +336,6 @@ hostapd_set_bss_options() {
 	set_default chan_util_avg_period 600
 	set_default utf8_ssid 1
 	set_default multi_ap 0
-	set_default rssi_reject_assoc_rssi 0
```
Comment (ArifAlam) [inline feeds/wifi-trunk/hostapd/files/hostapd.sh:341]: Why is rssi_reject_assoc_rssi getting removed?

Comment (dfranusic) [reply feeds/wifi-trunk/hostapd/files/hostapd.sh:341]: Because of the following commit: a9b3afd. It did not properly fix the issues with **hostapd** on **ecw5211** device.

**Comment by dfranusic** [comment]
> We try to follow the kernel coding style for the osync2.0 code that we write. Thanks !
Then maybe you should consider adding the following to the **wlan-repo**:
https://github.com/torvalds/linux/blob/master/.clang-format

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 3 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:5]: I agree with you on that. However, if you look at the **13-synthetic_client** patch you'll see that the **"dpp_network_probe.h"** was added in the same manner.  Wouldn't it be better if we all followed the same guidelines? In which **"src"** directory should the newly added files reside?

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 9 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1337]: Also seen in **"13-synthetic_client"** patch.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 10 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1360]: Agreed.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 9 above)
Comment (blogic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1337]: in that case 13-synthetic_client needs to also be fixed

Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:1337]: > in that case 13-synthetic_client needs to also be fixed
@AmmadRehmatCUS

File: feeds/wlan-ap/opensync/patches/11-protobuf-definitions (hunk 1 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/11-protobuf-definitions:29]: This is a result of one of my forcefully pushed commits. This should be removed, I agree.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 4 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:698]: What do you mean by this exactly? All the other patches (**16-dhcp-client.patch** for example) do the same.
```
Index: opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_network_probe.h
===================================================================
--- opensync-2.0.5.0.orig/src/lib/datapipeline/inc/dpp_network_probe.h
+++ opensync-2.0.5.0/src/lib/datapipeline/inc/dpp_network_probe.h
```

**Comment by dfranusic** [comment]
- code format issues (comments and tabs) fixed in the latest commit

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 3 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch]: I can fix this now if that's ok with the author. @AmmadRehmatCUS

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 4 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch:331]: I figured out what you were saying. Closing this.

File: feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch (hunk 3 above)
Comment (dfranusic) [reply feeds/wlan-ap/opensync/patches/17-dpp-sta-events.patch]: Fixed if 4041313

**Comment by blogic** [comment]
Hi,
can you fold all the fixes into the original patches ? that makes it easier to review
   John

**Comment by dfranusic** [comment]
Hi John

Just to clarify. You'd like me to squash all these commits into a single one?

Thanks.

**Comment by blogic** [comment]
@dfranusic well, they should be split up by feature or whoever it makes sense. what we dont want is a patch adding a feature and then another fixing that patch, which is what we have right now i think.

**Comment by dfranusic** [comment]
Oh, now I see. Yes, at the moment we have a couple of things included in this pull request:

1. build system update (opensync_stats.proto compiles during the build process)
2. new stats_type (**event**) for **Wifi_Stats_Config** (real-time event processing)
3. **clang-format** configuration file from Kernel source tree
4. fixes for new files that were added via patches instead of copied to the **src** directory of **opensync** package in **wlan-ap feed**

I'll try to reorganize my last 5 commits into these 4 logical units. In the end, you should have 4 commits to review.

**Comment by blogic** [comment]
thanks, sounds perfect

**Comment by dfranusic** [comment]
I believe this pull request should be closed since we are moving to a completely different approach due to **netlink** issues caused by hostapd's exclusivity over a certain subset of messages.
